docs/testing/README.md

```diff
@@ -0,0 +1,124 @@
+# Testing
+
+## How to test
+
+Follow the [Guiding Principles of React Testing Library](https://github.com/testing-library/react-testing-library#guiding-principles) when writing tests.
+
+## What to test
+
+### Accessibility
+
+Automated accessibility tests are required for significant component variations and should use existing `toHaveNoAxeViolations()` and `toHaveNoDAPViolations()` Jest matchers.
+
+Use this structure for automated accessibility tests:
+
+```
+test('should have no Axe or DAP violations', async () => {
+  const main = document.createElement('main');
+  render(
+    <Component />,
+    {
+      // DAP requires a landmark '<main>' in the DOM:
+      container: document.body.appendChild(main),
+    }
+  );
+
+  await expect(document.body).toHaveNoAxeViolations();
+  await expect(document.body).toHaveNoDAPViolations('ComponentName');
+});
+
+
+test('should have no Axe or DAP violations with component variation', async () => {
+  const main = document.createElement('main');
+  render(
+    <Component variationProp={value} />,
+    {
+      // DAP requires a landmark '<main>' in the DOM:
+      container: document.body.appendChild(main),
+    }
+  );
+
+  await expect(document.body).toHaveNoAxeViolations();
+  await expect(document.body).toHaveNoDAPViolations('ComponentName with variation');
+});
+```
+
+#### Test structure
+
+- Each significant variation of a component should have its own test block.
+- Use an **asynchronous** test block, and **await** each accessibility rule matcher.
+
+#### DAP requirements
+
+- DAP requires that components be rendered inside a valid landmark element, which is why the `container` is a `main` node. This extra step may not be required if the component is already wrapped in a `main` or another significant HTML landmark element.
+- DAP requires a unique id per test within a given component. (Hence, "ComponentName" and "ComponentName with variation")
+
+## User events
```

Comment:
👏Awesome! What other kinds of unit tests have you discovered that we need to consider documenting for component contributors?

Reply:
Keyboard events would be the other example here but personally I haven't had success using them. (Specifically I've wanted to simulate "Enter", "Spacebar", and arrow keys)

Reply:
We would also need to have a section on keyboard input (outside of "Enter", "Spacebar" and arrow keys used to traverse elements), which I think @diego-codes did for his `FilterSearch` tests.

Reply:
Good point! I think it'd be super helpful to show an example of testing consumer-facing props too, for example, `className` or `...other`